cmd/up/xpkg/batch.go

```diff
@@ -0,0 +1,569 @@
+	errPushPackageFmt     = "failed to push smaller provider package: %s"
+	errAbsAuthExtFmt      = "failed to get the absolute path for the authentication extension file: %s"
+	errReadAuthExtFmt     = "failed to read the authentication extension file at: %s"
+	errProcessFmt         = "\nfailed to process smaller provider package for %q"
+	errOutputAbsFmt       = "failed to get the absolute path for the package archive to store: %s/%s/%s"
+	errOpenPackageFmt     = "failed to open package file for writing: %s"
+	errWritePackageFmt    = "failed to store package archive in: %s"
+	errBatch              = "processing of at least one smaller provider has failed"
+)
+
+const (
+	wildcard  = "*"
+	tagLatest = "latest"
+)
+
+// AfterApply constructs and binds Upbound-specific context to any subcommands
+// that have Run() methods that receive it.
+func (c *batchCmd) AfterApply(kongCtx *kong.Context) error {
+	c.fs = afero.NewOsFs()
+	// NOTE(aru): we currently only support fetching family base image from
+	// daemon, but may opt to support additional sources in the future.
+	c.fetch = daemonFetch
+
+	upCtx, err := upbound.NewFromFlags(c.Flags)
+	if err != nil {
+		return err
+	}
+	kongCtx.Bind(upCtx)
+
+	return nil
+}
+
+// batchCmd builds and pushes a family of Crossplane provider packages.
+type batchCmd struct {
+	fs    afero.Fs
+	fetch fetchFn
+
+	FamilyBaseImage        string   `help:"Family image used as the base for the smaller provider packages." required:""`
+	ProviderName           string   `help:"Provider name, such as provider-aws to be used while formatting smaller provider package repositories." required:""`
+	FamilyPackageURLFormat string   `help:"Family package URL format to be used for the smaller provider packages. Must be a valid OCI image URL with the format specifier \"%s\", which will be substituted with <provider name>-<service name>." required:""`
+	Services               []string `help:"Services to build the scoped provider packages for." default:"monolith"`
+	Concurrency            uint     `help:"Maximum number of packages to process concurrently. Setting it to 0 puts no limit on the concurrency, i.e., all packages are processed in parallel." default:"0"`
+	PushRetry              uint     `help:"Number of retries when pushing a provider package fails." default:"3"`
+
+	Platform        []string `help:"Platforms to build the packages for. Each platform should use the <OS>_<arch> syntax. An example is: linux_arm64." default:"linux_amd64,linux_arm64"`
+	ProviderBinRoot string   `short:"p" help:"Provider binary paths root. Smaller provider binaries should reside under the platform directories in this folder." type:"existingdir"`
+	Output          string   `short:"o" help:"Path of the package output directory." optional:""`
+	StorePackage    []string `help:"Service name whose provider package should be stored under the package output directory specified with the --output option." optional:""`
```

Comment:
```suggestion
	StoreSubpackages    []string `help:"Package names whose providers should be stored under the package output directory specified with the --output-dir option." optional:""`
```
Also related to naming `Services` vs `Subpackages`, there seems to be an inconsistency in how we are naming these smaller provider groups.

Reply:
Done per our offline discussion.